tools/depends/target/Toolchain.cmake.in

```diff
@@ -80,3 +80,5 @@ set(ENV{PKG_CONFIG_LIBDIR} @prefix@/@deps_dir@/lib/pkgconfig)
 if(NOT CORE_SYSTEM_NAME STREQUAL linux)
   set(ADDONS_PREFER_STATIC_LIBS ON)
 endif()
+
+set(KODI_CROSSCOMPILING 1)
```

Comment:
even though osx uses the native compiler it's still a cross compile in every other sense. You could simply switch the target arch to 32bit (as we had some time ago) and it'd build without any other change.